lib/ts/fetch.ts

```diff
@@ -428,22 +433,24 @@ export async function onUnauthorisedResponse(
                 );
 
                 logDebugMessage("onUnauthorisedResponse: Refresh call ended");
-                let removeIdRefreshToken = true;
-                const idRefreshToken = response.headers.get("id-refresh-token");
-                if (idRefreshToken) {
-                    logDebugMessage(
-                        "onUnauthorisedResponse: Setting sIRTFrontend from refresh API call: " + idRefreshToken
-                    );
-                    await setIdRefreshToken(idRefreshToken, response.status);
-                    removeIdRefreshToken = false;
-                }
-                if (response.status === AuthHttpRequest.config.sessionExpiredStatusCode) {
-                    logDebugMessage("onUnauthorisedResponse: Refresh status code is: " + response.status);
-                    // there is a case where frontend still has id refresh token, but backend doesn't get it. In this event, session expired error will be thrown and the frontend should remove this token
-                    if (removeIdRefreshToken) {
-                        await setIdRefreshToken("remove", response.status);
-                    }
+
+                await saveTokensFromHeaders(response);
+
+                logDebugMessage("onUnauthorisedResponse: Refresh status code is: " + response.status);
+
+                const isUnauthorised = response.status === AuthHttpRequest.config.sessionExpiredStatusCode;
+
+                // There is a case where the FE thinks the session is valid, but backend doesn't get the tokens.
+                // In this event, session expired error will be thrown and the frontend should remove this token
```

Comment:
is this still possible? Can we eliminate this state somehow? If not, it's OK - but just asking

Reply:
This should be really rare:
1. someone clearing HTTP only cookies
2. some issue before/during saving the front-token
Still, we can't eliminate it completely